media_kit_video/lib/media_kit_video_controls/src/controls/methods/fullscreen.dart

```diff
@@ -22,12 +23,12 @@ Future<void> enterFullscreen(BuildContext context) {
         final stateValue = state(context);
         final contextNotiferValue = contextNotifier(context);
         final controllerValue = controller(context);
-        final controlsThemeDataBuilderValue = controlsThemeDataBuilder(context);
         Navigator.of(context, rootNavigator: true).push(
           PageRouteBuilder(
-            pageBuilder: (_, __, ___) =>
-                (controlsThemeDataBuilderValue ?? (e) => e).call(
-              VideoStateInheritedWidget(
+            pageBuilder: (_, __, ___) => VideoControlsThemeDataInjector(
+              // NOTE: Make various *VideoControlsThemeData from the parent context available in the fullscreen context.
+              context: context,
```

Comment:
The key is to pass the original `context` (non-fullscreen, where all `*VideoControlsTheme`s are present) to `VideoControlsThemeDataInjector` & it will make them available in the new fullscreen context (of different route).